src/main/java/edu/unc/lib/boxc/migration/cdm/services/PermissionsService.java

```diff
@@ -64,6 +72,54 @@ public void generateDefaultPermissions(PermissionMappingOptions options) throws
                         everyoneField,
                         authenticatedField);
             }
+
+            // works and files
+            if (options.isWithWorks() && options.isWithFiles()) {
```

Comment:
I think we could probably dry this up some. One approach could be to have a helper method named `queryForMappingIds(options)` which returns a list of ids (it could return an empty list if no appropriate options are set), and then use that list to add entries to the CSV.